### ISSUE TITLE
TreeGrid Item Clicklistener only fires for leaf nodes

### ISSUE THREAD

**Issue by aze-kool** [vaadin/vaadin-grid#1934]
https://vaadin.com/forum/thread/17773630/treegrid-itemclicklistener-not-working-properly

SelectionListner/ItemClicklistner are only called for leaf nodes and root nodes. for nodes having child nodes, the event is not fired. This is quite problematic for migration project. Since the tree component is not there and the only choice is the TreeGrid.

**Comment by mmoser18** [comment]
@tomivirkki: Thanks for this! It's a real life-saver! It saved me from abandoning Vaadin for my purpose!
My only question to the Vaadin team: why isn't this the default behavior? 
Trees should ALWAYS expand/collapse when clicking on the twisties only!

**Comment by johnthad** [comment]
> The work around provided above has a side effect: User can't select the text in the column because pointer-events: none
> TreeGrid should have option to turn off the click event on item text.

Yes. I'm using a tree grid to represent a folder so I'd like the toggle to open and close the folder while the text in the toggle to respond to double click. Can the `slot` where the toggle's `textContent` resides have a `part` so it could get pointer events?
